src/Bicep.Core/Syntax/IdentifierSyntax.cs

```diff
@@ -41,14 +46,53 @@ public string IdentifierName
                         return identifier.Text;
 
                     case SkippedTriviaSyntax skipped:
-                        return skipped.Elements.Any() ? LanguageConstants.ErrorName : LanguageConstants.MissingName;
+
+                        ImmutableArray<SyntaxBase> elements = skipped.Elements;
```

Comment:
>skipped [](http://example.com/codeflow?start=62&length=7)

This relies on a specific parser recovery behavior, which makes it kind of fragile. (We also want to avoid regexes in the main compiler path.)

If we agree that we need to support it in the parser (see my other comment), it should be its own type of syntax node and parsed by the parser like everything else.

Reply:
Thanks Marcin, I'll investigate separate syntax node option and see how it goes.